### PR TITLE
Shell plugin supports obtaining execution nodes through roles

### PR DESCRIPTION
**Pull request by Stevent-fei**

### CODE REVIEW

File: pkg/plugin/shell_plugin.go [hunk 1]
```diff
@@ -62,8 +63,12 @@ func (s Sheller) Run(context Context, phase Phase) error {
 				return fmt.Errorf("nodes is not found by label [%s]", on)
 			}
 			allHostIP = ipList
-		} else {
+		} else if apply.IsIPList(on) {
 			allHostIP = utils.DisassembleIPList(on)
+		}else {
+			if err := context.Cluster.GetIPSByRole(on); err != nil{
```
Comment (bxy4543) [inline pkg/plugin/shell_plugin.go:69]: `err = GetIPSByRole`?

**Review by bxy4543 — APPROVED** [review]

File: pkg/plugin/shell_plugin.go [hunk 2]
```diff
@@ -62,6 +62,11 @@ func (s Sheller) Run(context Context, phase Phase) error {
 				return fmt.Errorf("nodes is not found by label [%s]", on)
 			}
 			allHostIP = ipList
+		} else if on == common.MASTER || on == common.NODE {
+			role := context.Cluster.GetIPSByRole(on)
```
Comment (fanux) [inline pkg/plugin/shell_plugin.go:66]: So the return values is iplist not role.  role->iplist

Comment (Stevent-fei) [reply pkg/plugin/shell_plugin.go]: ok

File: pkg/plugin/shell_plugin.go [hunk 3]
```diff
@@ -62,6 +62,11 @@ func (s Sheller) Run(context Context, phase Phase) error {
 				return fmt.Errorf("nodes is not found by label [%s]", on)
 			}
 			allHostIP = ipList
+		} else if on == common.MASTER || on == common.NODE {
+			ipList := context.Cluster.GetIPSByRole(on)
```
Comment (fanux) [inline pkg/plugin/shell_plugin.go:66]: Where to use this ipList? Exec on allHostIP.

File: sealer/cmd/run.go [hunk 4]
```diff
@@ -53,7 +53,8 @@ create cluster to your baremetal server, appoint the iplist:
 	sealer run kubernetes:v1.19.8 --masters 192.168.0.2,192.168.0.3:23,192.168.0.4:24 \
 	--nodes 192.168.0.5:25,192.168.0.6:25,192.168.0.7:27 --passwd xxx
 create a cluster with custom environment variables:
-	sealer run -e DashBoardPort=8443 mydashboard:latest registry.cn-qingdao.aliyuncs.com/sealer-io/kubernetes:v1.19.8 --masters 3 --nodes 3
+	sealer run -e DashBoardPort=8443 mydashboard:latest registry.cn-qingdao.aliyuncs.com/sealer-io/kubernetes:v1.19.8 --masters 192.168.0.2,192.168.0.3,192.168.0.4 \
```
Comment (kakaZhou719) [inline sealer/cmd/run.go:56]: sealer run -e DashBoardPort=8443 mydashboard:latest 。no need to set "registry.cn-qingdao.aliyuncs.com/sealer-io/kubernetes:v1.19.8"

Comment (Stevent-fei) [reply sealer/cmd/run.go:56]: ok

File: pkg/plugin/shell_plugin.go [hunk 5]
```diff
@@ -62,6 +62,11 @@ func (s Sheller) Run(context Context, phase Phase) error {
 				return fmt.Errorf("nodes is not found by label [%s]", on)
 			}
 			allHostIP = ipList
+		} else if on == common.MASTER || on == common.NODE {
+			allHostIP := context.Cluster.GetIPSByRole(on)
```
Comment (fanux) [inline pkg/plugin/shell_plugin.go:66]: allHostIP := context.Cluster.GetIPSByRole(on)  ->  allHostIP = context.Cluster.GetIPSByRole(on)

**Comment by fanux** [comment]
Update the plugin docs

**Pull request merged**